debugging/netcoreapp/appStorage.ts

```diff
@@ -0,0 +1,63 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as path from 'path';
+import { FileSystemProvider } from "./fsProvider";
+
+export interface MementoAsync {
```

Comment:
`AppStorage` represents a state at the "application" or "project" level, which may often be sub-workspace-state.  (You might have several applications within the workspace folder, say, if using VS Code to open a solution created by VS comprised of several individual ASP.NET projects.)  It's intended to be used for things that are specific to the application (e.g. cache settings used to build/run its Docker images/containers) and not dependent on which folder was opened in VS Code (e.g. the solution level or project level folder) to edit that project.  This parallels how VS stores such data.

Reply:
Okay I'll have to try out the scenario you described. I just want to make sure we're following VS Code conventions here, which may be different than VS. In general, VS Code has a much bigger reliance on the fact that the project is at the root of the workspace. They also have a concept of 'multi-root' workspaces, which might be how they handle the multi-project scenario you're describing (which is handled appropriately by workspaceState)

Reply:
I think it would be good to have a meeting to discuss how we expect users to manage these types of projects/solutions in vscode.  Do we have actual customer usage data?